wlauto/workloads/googlephotos/__init__.py

```diff
@@ -0,0 +1,131 @@
+#    Copyright 2014-2016 ARM Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import wlauto.common.resources
+
+from wlauto import AndroidUiAutoBenchmark, Parameter
+from wlauto.exceptions import ValidationError
+from wlauto.utils.types import list_of_strings
+
+__version__ = '0.1.1'
+
+
+class Googlephotos(AndroidUiAutoBenchmark):
+
+    name = 'googlephotos'
+    package = 'com.google.android.apps.photos'
+    activity = 'com.google.android.apps.photos.home.HomeActivity'
+    view = [package + '/com.google.android.apps.consumerphotoeditor.fragments.ConsumerPhotoEditorActivity',
+            package + '/com.google.android.apps.photos.home.HomeActivity',
+            package + '/com.google.android.apps.photos.localmedia.ui.LocalPhotosActivity',
+            package + '/com.google.android.apps.photos.onboarding.AccountPickerActivity',
+            package + '/com.google.android.apps.photos.onboarding.IntroActivity']
+    description = """
+    A workload to perform standard productivity tasks with Google Photos.  The workload carries out
+    various tasks, such as browsing images, performing zooms, post-processing and saving a selected
+    image to file.
+
+    Test description:
+    1. Four images are copied to the device
+    2. The application is started in offline access mode
+    3. Gestures are performed to swipe between images and pinch zoom in and out of the selected
+       image
+    4. The Colour of a selected image is edited by selecting the colour menu, incrementing the
+       colour, resetting the colour and decrementing the colour using the seek bar.
+    5. A Crop test is performed on a selected image.  UiAutomator does not allow the selection of
+       the crop markers so the image is tilted positively, reset and then negatively to get a
+       similar cropping effect.
+    6. A Rotate test is performed on a selected image, rotating anticlockwise 90 degrees, 180
+       degrees and 270 degrees.
+    """
+
+    default_test_images = [
+        'uxperf_1200x1600.png', 'uxperf_1600x1200.jpg',
+        'uxperf_2448x3264.png', 'uxperf_3264x2448.jpg',
+    ]
```

Comment:
This was discussed earlier and the images will be hosted separately on a publicly-visible assets server (most likely a separate GitHub repo).